rewrite-java/src/main/java/org/openrewrite/java/nullability/StandardizeNullabilityAnnotationsVisitor.java

```diff
@@ -0,0 +1,326 @@
+        List<J.Annotation> cleanedAnnotations = new LinkedList<>(variableDeclarations.getLeadingAnnotations());
+        cleanedAnnotations.removeAll(currentNullabilityAnnotations);
+
+        J.VariableDeclarations cleanedVariableDeclaration = variableDeclarations.withLeadingAnnotations(cleanedAnnotations);
+        return annotationsForReplacement.stream().reduce(
+                cleanedVariableDeclaration,
+                (J.VariableDeclarations currentDeclaration, NullabilityAnnotation annotation) -> currentDeclaration.withTemplate(
+                        annotationTemplate(annotation),
+                        currentDeclaration.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName))
+                ), (oldDeclaration, newDeclaration) -> newDeclaration
+        );
+    }
+
+    private List<NullabilityAnnotation> getAnnotationsForReplacement(Set<MatchedNullabilityAnnotation> matchedNullabilityAnnotations, Set<NullabilityAnnotation.Target> possibleTargetTypes) {
+        Set<NullabilityAnnotation> usedNullabilityAnnotations = matchedNullabilityAnnotations.stream().map(MatchedNullabilityAnnotation::getNullabilityAnnotation).collect(Collectors.toSet());
+        Set<NullabilityAnnotation.Nullability> matchedNullabilities = usedNullabilityAnnotations.stream().map(NullabilityAnnotation::getNullability).collect(Collectors.toSet());
+        if (matchedNullabilities.isEmpty()) {
+            // no nullabilities, nothing to do
+            return emptyList();
+        }
+        if (matchedNullabilities.size() > 1) {
+            // different nullabilities on a single element, we better do nothing
+
+            // TODO log useful information
+            // Thought:
+            // Maybe we should transfer the logging to the caller
+            // as that code knows more about the element we failed at
+            return emptyList();
+        }
+
+        NullabilityAnnotation.Nullability nullability = matchedNullabilities.stream().findFirst().orElse(null);
+        Set<NullabilityAnnotation.Scope> scopesToCover = usedNullabilityAnnotations
+                .stream()
+                .map(NullabilityAnnotation::getScopes)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+        if (scopesToCover.isEmpty()) {
+            return emptyList();
+        }
+
+        List<NullabilityAnnotation> annotationsForReplacement = getAnnotationsForReplacement(nullability, possibleTargetTypes, scopesToCover);
+        if (annotationsForReplacement.isEmpty()) {
+            // We were not able to find a good replacement.
+
+            // TODO log useful information
+            // Thought:
+            // Maybe we should transfer the logging to the caller
+            // as that code knows more about the element we failed at.
+        }
+        if (annotationsForReplacement.containsAll(usedNullabilityAnnotations) && usedNullabilityAnnotations.containsAll(annotationsForReplacement)) {
```

Comment:
You are right, it should be more efficient to compare sets instead of this workaround. I will change as you proposed, because I would like to maintain order. This way we replace annotations in a deterministic manner. Another advantage ist, that you can define the "priority" for replacement inside configuration. E. g. if you do not use a library which provides all nullability annotations you need, you could also configure the recipe to use a combination of ``Jetbrains` and a second library which has an annotation that also applies on package-level while ensuring that we use `Jetbrains` where possible.